mdata/store_cassandra.go

```diff
@@ -435,12 +449,20 @@ func (c *CassandraStore) SearchTable(key, table string, start, end uint32) ([]ch
 	results := make(chan outcome, numQueries)
 	for i := range crrs {
 		crrs[i].out = results
-		c.readQueue <- crrs[i]
+		select {
+		case c.readQueue <- crrs[i]:
+		default:
+			cassReadQueueFull.Inc()
+			return nil, errReadQueueFull
+		}
```

Comment:
so the value of read queue size now becomes critically important. here's how I think we can figure out a sane setting:
* for any query, for any target retrieved, 1 CRR is issued per month of data crossed, irrespective of which aggregation we're using.
* the read queue should allow to buffer a temporay peak in workload, as long as that workload can be cleared out of the queue in, say, 10 seconds.
* thus, at any given time, we need to allow to put 10 seconds worth of the max-handleable workload in the queue.
* it's better to over-estimate what the max-handleable workload is (because we'll just drop reads for being too old from the queue) than it is to under-estimate (which would result in a smaller queue which is full too quickly, resulting in render requests erroring whereas they could've been processed fine)
* so let's aim for a rather high estimate, based on some numbers we've seen in the wild.
Let's say a heavy query involves 20k targets and each targets reads 2y=24months worth of data. let's say we (aim to) support 200 such requests per second. That's 96M CRR's per second, 960M for 10 seconds worth.
* for this kind of workload, you'd use >= 8 MT instances.
* so each instance would need a read queue that can hold  960M/8=120M CRR's.
* the queue is currently a buffered chan of pointers, so 120M*8B per pointer  = 960MB RAM worth of read queue.

So, this is one way of calculating how much resources we need for a read queue to accomodate legitimate CRR's (CRR's that we don't want dropped unless we really can't handle the load)

There's other ways to compute this of course.
For example, we also see in our dashboards that some clusters do up to 200k cassandra gets/s . (CRR's/sec), and this with healthy response times.
So we have to ask, how much higher should the volume be before we can confidently drop the reads before even giving them a chance to go through the queue (and be dropped if turns out it's too old). let's say a workload 20x that.that would be 4M CRR's/se, or over 10 seconds, that's 40M, or per node 40M/8=5M.  With this model, we're estimating 24x less than with the previous method.

So we can meet somewhere half way between the two methods, and suggest 10M read queue per node.

Here's something interesting though: looking at our dashboards, we have chunks of about 40B avg size, and the avg search results in less than 1 chunk per read; that means that per CRR we get no more than 40B of data.
the ChunkReadRequest type is 80 Bytes big (see `sizeof` utility for golang), and a pointer to it is 64bit, so each pending CRR costs us 88B of RAM, more than twice the amount of data we will actually retrieve !?

This is a reason why i'm not a huge fan of drop-on-full-queue.